wordpress-mobile/WordPress-iOS
# Reader Comments: Follow conversation via notifications

Refs: `pctCYC-9x-p2`
Depends on #17369

This change enables the "Follow Conversation via Notifications" feature flag for the public. Additionally, this PR also addresses https://github.com/wordpress-mobile/WordPress-iOS/pull/17339#discussion_r732240314 regarding localization strings placements.

For now, the strings are encapsulated in an enum in Swift, and exposed to the Obj-C side. Since the feature flag is not yet removed, it only returns the titles for enabling/disabling notifications. Once the feature flag is removed, I'll add the titles for following/unfollowing conversations.

## To test

- Go to Reader, and select any unsubscribed post.
- Verify that the "Follow" button should be visible on the navigation bar.
- Tap the Follow button.
- Verify that the snackbar with the "Enable" action button is shown.
- Tap on the "Enable" button.
- Verify that a feedback notice with an "Undo" action button is shown.
- Tap on the "Undo" action button.
- Verify that a feedback notice is shown.
- Verify that the Follow button has turned into a bell icon button in the navigation bar.
- Go to Reader, and select any subscribed post.
- Tap on the bell icon button.
- Verify that the bottom sheet is shown.
- Tap on the switch.
- Dismiss the bottom sheet.
- Verify that a feedback notification is shown, and correctly represents the desired state.
- Tap the bell icon button again.
- Tap on the Unfollow Conversation button.
- Verify that a feedback notification is shown.
- Verify that the bell icon button has now turned back into "Follow".

## Related PRs:
- #17369
- #17340 
- #17339 
- #17333
- #17322
- https://github.com/wordpress-mobile/WordPressKit-iOS/pull/452

## Regression Notes
1. Potential unintended areas of impact
When updating from the previous version, the "follow conversation" state may be incorrectly represented (e.g. off when it's supposed to be on). Otherwise, the feature itself is pretty isolated so it should not impact other features.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the feature.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for the service changes related to the follow conversation feature.

PR submission checklist:
